lib/apollo-federation/has_directives.rb

```diff
@@ -2,11 +2,20 @@
 
 module ApolloFederation
   module HasDirectives
-    attr_reader :federation_directives
-
     def add_directive(name:, arguments: nil)
-      @federation_directives ||= []
-      @federation_directives << { name: name, arguments: arguments }
+      own_federation_directives << { name: name, arguments: arguments }
+    end
+
+    def federation_directives
```

Comment:
Would you mind updating this implementation to use `GraphQL::Schema::FindInheritedValue`?